src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -30,19 +31,20 @@ use crate::hummock::compactor::task_progress::TaskProgress;
 use crate::hummock::iterator::{Forward, HummockIterator};
 use crate::hummock::sstable_store::{BlockStream, SstableStoreRef};
 use crate::hummock::value::HummockValue;
-use crate::hummock::{Block, BlockHolder, BlockIterator, HummockResult};
+use crate::hummock::{BlockHolder, BlockIterator, HummockResult, TableHolder};
 use crate::monitor::StoreLocalStatistic;
 
 /// Iterates over the KV-pairs of an SST while downloading it.
-struct SstableStreamIterator {
+pub struct SstableStreamIterator {
+    sstable_store: SstableStoreRef,
+    table_holder: TableHolder,
```

Comment:
The table_holder seems a refactor/optimization that is orthogonal to recreating stream. I think it helps reduce IO but risk OOM if SST metadata is large.

I'd +1 for this refactor. @Li0k

Reply:
Estimated size might cover this, but when the sst meta takes up most of the space of the entire object (> estimated size), then oom can happen.

I don't have any other ideas about this other than reducing max_key_count to reduce the number of tombstones per meta. cc @zwang28 @Little-Wallace